### ISSUE TITLE
Hide empty folders does not work with root shared folders

### ISSUE THREAD

**Issue by 444nonymous** [UniversalMediaServer/UniversalMediaServer#1378]
If the folders listed in "shared folders" are empty, they should not show when "Hide empty folders" is selected. Right now it only works with subfolders of the root folders.

The only problem could be not showing any shared folder in the client. But if there are not folders with any content, why should they be shown? Just a waste of time.

So i think "hide empty folders" should apply to any folder, not just subfolders.

**Comment by Nadahar** [comment]
I've never tried "hide empty folder" but what you say is reasonable. I don't think there's any easy way to hide the shared folder itself though.

Regarding the sharing of everything when nothing is shared, it *is* intentional. Some users don't know where they keep their files or don't understand the concept of files and folders itself. This makes sure that even basic users that doesn't configure their shared folders will share content.

Personally I don't find it very intuitive that "nothing" means "all", but it's the way it's always been. The bottom line is that we need a default config that will share some content, or we will have an endless stream of support threads that complains "Can't play UMS on my TV" and similar.

**Comment by SubJunk** [comment]
@444nonymous I agree that "Hide empty folders" would ideally work for root folders too, it would be a good change

@Nadahar agreed it's a good default, and I think our GUI makes it clear that all folders are shared if the user goes to that tab.

We could slightly change the text at the end of the first time startup wizard from:

> You can optionally specify your shared folders on the Navigation/Share Settings tab

to something like

> All of your folders are now shared. You can optionally specify your shared folders on the Navigation/Share Settings tab
